packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1371,11 +1407,25 @@ export class KeyringController extends BaseController<
   /**
    * Verifies the that the seed phrase restores the current keychain's accounts.
    *
+   * @param keyringId - The id of the keyring to verify.
    * @returns Promise resolving to the seed phrase as Uint8Array.
    */
-  async verifySeedPhrase(): Promise<Uint8Array> {
+  async verifySeedPhrase(keyringId?: string): Promise<Uint8Array> {
     this.#assertIsUnlocked();
-    return this.#withControllerLock(async () => this.#verifySeedPhrase());
+    const keyring = this.#getKeyringByIdOrDefault(keyringId);
+
+    if (!keyring) {
+      throw new Error(KeyringControllerError.NoHdKeyring);
+    }
```

Comment:
This error message could be confusing because it can be thrown whenever a non-HD ID is passed to this function, which doesn't necessarily mean that there is no HD keyring. This could be easily confused with an empty vault case, which is the reason this error was originally added [here](https://github.com/MetaMask/core/blob/4fbbd1309c4cbba2c363e235946c9c074c611eaf/packages/keyring-controller/src/KeyringController.ts#L2196-L2196)

Can we throw a more specific error message?

Reply:
Also, thoughts on moving all this logic to the internal `#verifySeedPhrase` so that it is applied to internal caller as well?

Reply:
Sounds like a good idea, so we won't repeat the code and will be easier to test in unit tests